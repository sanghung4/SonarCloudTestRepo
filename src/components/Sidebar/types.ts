export interface NavigationTypes {
  name: string;
  href: string;
  group: string;
  icon: (props: React.ComponentProps<"svg">) => JSX.Element;
}
