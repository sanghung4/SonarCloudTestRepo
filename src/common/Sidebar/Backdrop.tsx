import { Box, Fade, SxProps } from '@dialexa/reece-component-library';

type Props = {
  on: boolean;
  onClick: () => void;
  sx?: SxProps;
};

export default function Backdrop(props: Props) {
  return (
    <Fade in={props.on}>
      <Box
        data-testid="branch-sidebar-backdrop"
        onClick={props.onClick}
        zIndex={1}
        position="absolute"
        left={0}
        right={0}
        width={1}
        height={1}
        sx={{ backgroundColor: 'rgba(0, 0, 0, 0.5)', ...props.sx }}
      />
    </Fade>
  );
}
