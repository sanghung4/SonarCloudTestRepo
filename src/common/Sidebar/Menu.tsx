import { ReactNode } from 'react';

import clsx from 'clsx';
import { useLocation, useNavigate } from 'react-router-dom';

import { Button } from 'components/Button';
import { ReactComponent as ListIcon } from 'resources/icons/list.svg';
import { ReactComponent as UsersIcon } from 'resources/icons/users.svg';

/**
 * Types
 */
type MenuItem = {
  icon: ReactNode;
  name: string;
  path: string;
};

/**
 * Config
 */
export const menuItems: MenuItem[] = [
  { icon: <UsersIcon />, name: 'Existing Customers', path: '/' },
  { icon: <ListIcon />, name: 'All Catalogs', path: '#' }
];

/**
 * Component
 */
function Menu() {
  /**
   * Custom Hooks
   */
  const { pathname } = useLocation();
  const navigate = useNavigate();

  /**
   * Render
   */
  return (
    <div>
      {menuItems.map((item, i) => {
        const selected = item.path === pathname;
        return (
          <Button
            className={clsx('w-full text-left', {
              'bg-primary-2-80': selected,
              'text-secondary-2-30': !selected,
              'text-white': selected,
              'font-normal': !selected,
              'font-bold': selected
            })}
            data-testid={`sidebar-menu_item${i}`}
            iconPosition="left"
            icon={item.icon}
            align="justify-start"
            key={`nav-item${i}`}
            title={item.name}
            onClick={() => navigate(item.path)}
          />
        );
      })}
    </div>
  );
}
export default Menu;
