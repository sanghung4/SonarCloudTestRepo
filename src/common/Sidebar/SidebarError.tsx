import { Box, Typography, styled } from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import leak from 'images/leak.svg';
import leak_bg from 'images/leak_bg.svg';

const ImgStyled = styled('img')(() => ({
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  height: '60%',
  '&:first-child': {
    height: '100%'
  }
}));

function SidebarError() {
  /**
   * Custom Hooks
   */
  const { t } = useTranslation();

  /**
   * Render
   */
  return (
    <Box
      px={4}
      pt={10}
      display="flex"
      flexDirection="column"
      alignItems="center"
    >
      <Box position="relative" width="171px" height="195px">
        <ImgStyled src={leak_bg} alt="leaky pipe" />
        <ImgStyled src={leak} alt="leaky pipe" />
      </Box>
      <Typography
        variant="h1"
        pt={3}
        pb={1}
        color="primary02.main"
        fontWeight={400}
      >
        {t('common.ohNo')}
      </Typography>
      <Typography variant="h4">{t('common.leak')} </Typography>
      <Box py={2}>
        <Typography variant="body1" align="center">
          {t('branch.error')}
        </Typography>
      </Box>
    </Box>
  );
}
export default SidebarError;
