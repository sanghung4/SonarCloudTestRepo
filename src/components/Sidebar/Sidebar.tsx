import { useState, Fragment, useEffect } from "react";
import { MenuIcon, DotsHorizontalIcon } from "@heroicons/react/solid";
import { LogoutIcon, UserIcon } from "@heroicons/react/outline";
import { NavLink } from "react-router-dom";
import clsx from "clsx";
import { Menu, Transition } from "@headlessui/react";
import { Login } from "../../pages/Login";
import { useOktaAuth } from "@okta/okta-react";
import { SvgList } from "../SvgList";
import { navigation } from "./utils";
import { useAuthContext } from "../../store/AuthProvider";
import { empty } from "@apollo/client";

const Sidebar = () => {
  const { authState, oktaAuth } = useOktaAuth();
  const { userInfo, userBranchId } = useAuthContext();

  const [sideBar, toggleSideBar] = useState(false);
  const [showMore, setShowMore] = useState(false);

  const logOut = () => oktaAuth.signOut();

  if (!authState || !authState.isAuthenticated) {
    return <Login />;
  }

  // BRANCH LISTING CONFIG
  const showMoreVisibility = userBranchId ? userBranchId.length > 3 : false;

  const branchText = showMore || !showMoreVisibility ? userBranchId?.join(', '): userBranchId?.join(', ').substring(0, 20) + "...";

  const handleClick = () => {
    setShowMore ((prevShowMore) => !prevShowMore);
  }

  const menu = (
    <Menu as='div' className='relative inline-block text-left'>
      <div>
        <Menu.Button className='rounded-full flex items-center p-1 text-white hover:bg-reece-500 focus:none'>
          <DotsHorizontalIcon className='h-5 w-5' aria-hidden='true' />
        </Menu.Button>
      </div>
      <Transition
        as={Fragment}
        enter='transition ease-out duration-100'
        enterFrom='transform opacity-0 scale-95'
        enterTo='transform opacity-100 scale-100'
        leave='transition ease-in duration-75'
        leaveFrom='transform opacity-100 scale-100'
        leaveTo='transform opacity-0 scale-95'
      >
        <Menu.Items className='absolute bottom-8 right-0 mt-2 w-40 lg:w-48 rounded-md shadow-lg bg-white ring-1 ring-black ring-opacity-5 focus:outline-none'>
          <div className='py-1'>
            <Menu.Item>
              {({ active }: { active: boolean }) => (
                <button
                  onClick={logOut}
                  className={clsx(
                    active ? "bg-background text-gray-700" : "text-gray-700",
                    "flex w-full text-left px-4 py-2 text-sm"
                  )}
                >
                  <LogoutIcon className='w-5 h-5 text-gray-700 md:w-6 md:h-6 mr-2' />
                  Logout
                </button>
              )}
            </Menu.Item>
          </div>
        </Menu.Items>
      </Transition>
    </Menu>
  );

  return (
    <div>
      {/* mobile menu bar */}
      <div className='bg-primary-1-100 text-gray-50 flex justify-between lg:hidden'>
        <SvgList
          background='#003766'
          fill='white'
          className='h-14 p-4'
          name='Logo'
        />
        <button
          className='focus:bg-reece-600 p-4'
          onClick={() => toggleSideBar(!sideBar)}
        >
          <MenuIcon className='h-6 w-6' />
        </button>
      </div>

      {/* sidebar */}
      <div
        className={clsx(
          sideBar ? "translate-x-0" : "-translate-x-full",
          "overflow-hidden h-full flex-col flex px-2 bg-primary-1-100 w-44 lg:w-64 z-10 fixed transform inset-y-0 left-0 lg:translate-x-0 transition duration-200 ease-in-out"
        )}
      >
        <div className='w-full flex flex-col lg:flex-row justify-center items-center h-28 lg:h-24'>
          <SvgList
            background='#003766'
            fill='white'
            className='h-6'
            name='Logo'
          />
          <div className='text-white text-sm font-bold text-center pt-2 lg:pl-4 lg:pt-0'>
            COMPANION PORTAL
          </div>
        </div>
        <div className='flex flex-col justify-between h-full'>
          <nav>
            <ul className='mt-8'>
              {navigation.map(
                (item, index) =>
                  userInfo?.groups?.includes(item.group) && (
                    <li key={index}>
                      <NavLink
                        className={(isActive) => {
                          return clsx(
                            "flex",
                            "w-full",
                            "transition",
                            "duration-200",
                            "hover:bg-reece-600",
                            "hover:text-reece-50",
                            "cursor-pointer",
                            "items-center",
                            "text-sm",
                            "lg:text-base",
                            "mb-2",
                            "px-2",
                            "py-2",
                            "lg:mb-4",
                            "lg:px-6",
                            "lg:py-2",
                            "rounded",
                            isActive
                              ? "text-white bg-reece-500"
                              : "text-gray-400"
                          );
                        }}
                        to={item.href}
                      >
                        <item.icon className='w-5 h-5 md:w-6 md:h-6 color-white mr-2' />
                        {item.name}
                      </NavLink>
                    </li>
                  )
              )}
            </ul>
          </nav>
          <div className='flex-shrink-0 w-full group block lg:text-base mb-4 py-2 lg:mb-6 lg:px-6 rounded'>
            <div className='flex items-center justify-between'>
              <div className='flex items-center'>
                <UserIcon className='h-6 w-6 min-w-fit text-white' />
                <p className='text-xs font-small lg:text-base text-white ml-2'>
                  {userInfo?.name}
                </p>
              </div>
              <div>{menu}</div>
            </div>
            <div className='text-xs font-small text-white ml-8'>
              {userBranchId &&<p>{branchText}</p>}
              {showMoreVisibility && <p className="underline hover:cursor-pointer" onClick={() => handleClick()}>{showMore ? "See less" : "See more"}</p>}
            </div>
          </div>
        </div>
      </div>
      <div className='hidden lg:block w-64'></div>
    </div>
  );
};

export default Sidebar;
