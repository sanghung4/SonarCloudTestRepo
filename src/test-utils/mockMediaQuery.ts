import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
