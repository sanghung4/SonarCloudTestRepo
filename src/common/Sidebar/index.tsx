import { HTMLAttributes, useEffect, useMemo, useRef, useState } from 'react';

import {
  Box,
  IconButton,
  Modal,
  Slide,
  Typography,
  alpha,
  useScreenSize,
  SxProps
} from '@dialexa/reece-component-library';
import useResizeObserver from 'use-resize-observer/polyfilled';

import Backdrop from 'common/Sidebar/Backdrop';
import SidebarError from 'common/Sidebar/SidebarError';
import ConditionalWrapper from 'common/ConditionalWrapper';
import { CloseIcon } from 'icons';

type ElementProps = HTMLAttributes<HTMLElement>;
interface Props extends ElementProps {
  close: () => void;
  error?: boolean;
  on: boolean;
  title?: string;
  isContentLoading?: boolean;
  widthOverride?: number;
  containerSx?: SxProps;
  backdropSx?: SxProps;
}

function Sidebar(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();

  /**
   * State
   */
  const [sidebarHeight, setSidebarHeight] = useState(0);

  /**
   * Refs
   */
  const contentRef = useRef<HTMLDivElement>(null);
  const { height: contentHeight = 0 } = useResizeObserver<HTMLDivElement>({
    ref: contentRef
  });

  /**
   * Memo
   */
  const availableHeight = useMemo(availableHeightMemo, [
    contentHeight,
    contentRef
  ]);

  /**
   * Effect
   */
  useEffect(sidebarHeightEffect, [
    availableHeight,
    contentHeight,
    props.isContentLoading,
    props.on,
    isSmallScreen
  ]);
  useEffect(disableScrolling, [props.on]);

  /**
   * Render
   */
  return (
    <>
      <Backdrop
        on={props.on}
        onClick={props.close}
        sx={{ backdropFilter: 'none', ...props.backdropSx }}
      />
      <Box
        width="1"
        height={1}
        position="fixed"
        zIndex={-1}
        component="div"
        ref={contentRef}
      />
      <ConditionalWrapper
        condition={isSmallScreen}
        wrapper={(children: JSX.Element) => (
          <Modal open={props.on} disablePortal>
            {children}
          </Modal>
        )}
      >
        <Slide
          direction="left"
          in={props.on}
          mountOnEnter
          unmountOnExit
          children={
            <Box
              position="absolute"
              top={0}
              right={0}
              zIndex={3}
              overflow="hidden"
              bgcolor="common.white"
              boxShadow={(theme) =>
                `0 5px 20px 0 ${alpha(theme.palette.common.black, 0.1)}`
              }
              sx={{
                height: isSmallScreen ? '100vh' : undefined,
                maxHeight: isSmallScreen
                  ? undefined
                  : availableHeight || '100%',
                width: isSmallScreen ? '100vw' : props.widthOverride ?? 380,
                ...props.containerSx
              }}
            >
              <Box
                data-testid="branch-list"
                display="flex"
                flexDirection="column"
              >
                <Box
                  id="sidebar-header"
                  py={1}
                  pl={3}
                  pr={2}
                  display="flex"
                  justifyContent="space-between"
                  alignItems="center"
                  sx={(theme) => ({
                    boxShadow: `0 2px 10px 0 ${alpha(
                      theme.palette.primary03.main,
                      0.15
                    )}`
                  })}
                >
                  {!!props.title && (
                    <Typography variant="h5" color="primary">
                      {props.title}
                    </Typography>
                  )}
                  <IconButton
                    onClick={props.close}
                    size="large"
                    sx={{ p: 1 }}
                    data-testid="slider-close"
                  >
                    <CloseIcon />
                  </IconButton>
                </Box>
                {props.error ? (
                  <SidebarError />
                ) : (
                  <Box
                    sx={{
                      [isSmallScreen || props.isContentLoading
                        ? 'height'
                        : 'maxHeight']: sidebarHeight
                    }}
                    display="flex"
                    flexDirection="column"
                  >
                    {props.children}
                  </Box>
                )}
              </Box>
            </Box>
          }
        />
      </ConditionalWrapper>
    </>
  );

  /**
   * Memo defs
   */
  function availableHeightMemo() {
    return Math.min(
      contentHeight,
      window.innerHeight - (contentRef.current?.getBoundingClientRect().y ?? 0)
    );
  }

  /**
   * Effect defs
   */
  function sidebarHeightEffect() {
    if (!props.on || props.isContentLoading) {
      // 56 is estimated header height
      setSidebarHeight(
        (isSmallScreen ? window.innerHeight : contentHeight) - 56
      );
    } else {
      // recalculate the header height after it mounts
      (async () => {
        // (a)wait for sidebar to mount
        await new Promise((res) => setTimeout(res, 0));
        setSidebarHeight(
          (isSmallScreen ? window.innerHeight : availableHeight) -
            // Ignoreing since this will never occur in unit test
            // istanbul ignore next
            (document.getElementById('sidebar-header')?.getBoundingClientRect()
              .height ?? 0)
        );
      })();
    }
  }
  function disableScrolling() {
    const root = document.getElementById('root');
    // istanbul ignore next
    if (root) {
      root.style.overflow = props.on ? 'hidden' : 'visible';
    }
    // istanbul ignore next
    return () => {
      if (root) {
        root.style.overflow = 'visible';
      }
    };
  }
}

export default Sidebar;
