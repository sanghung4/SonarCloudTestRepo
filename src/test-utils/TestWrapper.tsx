import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
