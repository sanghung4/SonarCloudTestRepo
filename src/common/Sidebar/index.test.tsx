import { act } from 'react-dom/test-utils';

import Sidebar from 'common/Sidebar';
import Backdrop from 'common/Sidebar/Backdrop';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

const mockChild = <div>test</div>;

describe('common - Sidebar', () => {
  it('expect to match snapshot with basic props', () => {
    setBreakpoint('desktop');
    const { container } = render(
      <Sidebar close={jest.fn()} on>
        {mockChild}
      </Sidebar>
    );
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot with error', () => {
    setBreakpoint('desktop');
    const { container } = render(
      <Sidebar close={jest.fn()} on error>
        {mockChild}
      </Sidebar>
    );
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on desktop', async () => {
    setBreakpoint('desktop');
    const { container } = render(
      <Sidebar close={jest.fn()} on>
        {mockChild}
      </Sidebar>
    );
    await act(() => new Promise((res) => setTimeout(res, 10)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on mobile', async () => {
    setBreakpoint('mobile');
    const { container } = render(
      <Sidebar close={jest.fn()} on>
        {mockChild}
      </Sidebar>
    );
    await act(() => new Promise((res) => setTimeout(res, 10)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot for Backdrop', () => {
    setBreakpoint('desktop');
    const { container } = render(<Backdrop onClick={jest.fn()} on />);
    expect(container).toMatchSnapshot();
  });
});
