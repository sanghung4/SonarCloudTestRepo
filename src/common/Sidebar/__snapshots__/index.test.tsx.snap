// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common - Sidebar expect to match snapshot for Backdrop 1`] = `
<div>
  <div
    class="MuiBox-root css-1185gzh"
    data-testid="branch-sidebar-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
</div>
`;

exports[`common - Sidebar expect to match snapshot on desktop 1`] = `
<div>
  <div
    class="MuiBox-root css-rim9vd"
    data-testid="branch-sidebar-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    class="MuiBox-root css-96jqp"
  />
  <div
    class="MuiBox-root css-1o80w2w"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiBox-root css-j7qwjs"
      data-testid="branch-list"
    >
      <div
        class="MuiBox-root css-14cad6o"
        id="sidebar-header"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-tbe422-MuiButtonBase-root-MuiIconButton-root"
          data-testid="slider-close"
          tabindex="0"
          type="button"
        >
          <svg>
            delete.svg
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-9udc97"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - Sidebar expect to match snapshot on mobile 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiBox-root css-rim9vd"
    data-testid="branch-sidebar-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    class="MuiBox-root css-96jqp"
  />
  <div
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiBox-root css-o5ig29"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiBox-root css-j7qwjs"
        data-testid="branch-list"
      >
        <div
          class="MuiBox-root css-14cad6o"
          id="sidebar-header"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-tbe422-MuiButtonBase-root-MuiIconButton-root"
            data-testid="slider-close"
            tabindex="0"
            type="button"
          >
            <svg>
              delete.svg
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-vwn1lt"
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`common - Sidebar expect to match snapshot with basic props 1`] = `
<div>
  <div
    class="MuiBox-root css-rim9vd"
    data-testid="branch-sidebar-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    class="MuiBox-root css-96jqp"
  />
  <div
    class="MuiBox-root css-1o80w2w"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiBox-root css-j7qwjs"
      data-testid="branch-list"
    >
      <div
        class="MuiBox-root css-14cad6o"
        id="sidebar-header"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-tbe422-MuiButtonBase-root-MuiIconButton-root"
          data-testid="slider-close"
          tabindex="0"
          type="button"
        >
          <svg>
            delete.svg
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-9udc97"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - Sidebar expect to match snapshot with error 1`] = `
<div>
  <div
    class="MuiBox-root css-rim9vd"
    data-testid="branch-sidebar-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    class="MuiBox-root css-96jqp"
  />
  <div
    class="MuiBox-root css-1o80w2w"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiBox-root css-j7qwjs"
      data-testid="branch-list"
    >
      <div
        class="MuiBox-root css-14cad6o"
        id="sidebar-header"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-tbe422-MuiButtonBase-root-MuiIconButton-root"
          data-testid="slider-close"
          tabindex="0"
          type="button"
        >
          <svg>
            delete.svg
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-1307ysn"
      >
        <div
          class="MuiBox-root css-kfl6ja"
        >
          <img
            alt="leaky pipe"
            class="css-1r9pf21"
            src="leak_bg.svg"
          />
          <img
            alt="leaky pipe"
            class="css-1r9pf21"
            src="leak.svg"
          />
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-83vxwx-MuiTypography-root"
        >
          Oh no!
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-rbn354-MuiTypography-root"
        >
          Looks like we have a leak
           
        </h4>
        <div
          class="MuiBox-root css-1sf3xto"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1v8ps1j-MuiTypography-root"
          >
            Please refresh your browser or check back in a few minutes.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
